data-access-layer/metering/MeteringClient.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const config = require('../../common/config');
+const utils = require('../../common/utils');
+const HttpClient = utils.HttpClient;
+const logger = require('../../common/logger');
+
+class MeteringClient extends HttpClient {
+
+  constructor(options) {
+    super(_.defaultsDeep({
+      baseUrl: config.metering.metering_url,
+      headers: {
+        Accept: 'application/json',
+      },
+      followRedirect: false
+    }, options));
+    this.clientid = config.metering.client_id;
+    this.clientsecret = config.metering.client_secret;
+    this.token_url = config.metering.token_url;
+    this.metering_url = config.metering.metering_url;
+  }
+
+  getAuthToken() {
+    return this
+      .request({
+        baseUrl: this.token_url,
+        url: '/oauth/token',
+        auth: {
+          user: this.clientid,
+          pass: this.clientsecret
+        },
+        qs: {
+          grant_type: 'client_credentials'
+        }
+      }, 200)
+      .then(res => {
+        logger.debug('Metering auth response body', res.body);
+        const serverResponse = JSON.parse(res.body);
+        return serverResponse.access_token;
+      })
+      .catch(err => {
+        logger.error('Error occurred while fetching metering auth token', err);
+        throw err;
+      });
+  }
+
+  putUsageRecord(usage_records) {
```

Comment:
`usageRecords` ?

Reply:
Also, I am not sure what the method does. Does it do start and stop metering both?

Reply:
Hi please see:
https://wiki.wdf.sap.corp/wiki/display/CPC15N/Developer+Guide+for+Metering-as-a-Service
section `Usage Recording`
This is the API endpoint to send usage records which can have both stop and start.
`usageRecords` since we can send an object with multiple usages

Reply:
ah! Thanks!